### DISCUSSION TITLE
LDAP stops working after about a day

### DISCUSSION THREAD

**Discussion by SerWax** [rustdesk/rustdesk-server-pro#401] [General]
Hi

we've configured LDAP login, but it only works for a short while after setting it up. After about a day, LDAP logins fail with incorrect username/password, but no connections can be observer from the rustdesk pro server and the DCs. Going back to the configuration and changing something, wakes it up and logins start working again, only to stop after some time.

Is anyone else having this problem?

**Comment by rustdesk** [comment]
@fufesou

**Comment by fufesou** [comment]
@SerWax Hi, thanks for the feedback.

> After about a day, LDAP logins fail with incorrect username/password, but no connections can be observer from the rustdesk pro server and the DCs.

I think the problem you described is because a cache has expired. But I see no cache in LDAP auth.
I will continue testing.

Can you please
1. Give a screenshot of the error
2. Confirm that repeated login and logout works just after chaning something in LDAP configuration

**Comment by SerWax** [comment]
![{A8DABF86-30A8-4E7C-82BA-A1D407C3209B}](https://github.com/user-attachments/assets/d8fc698a-c6a3-4301-90e5-ef809fa432cc)
![{392AA634-72AD-40B8-B8E5-DFABC1D1B259}](https://github.com/user-attachments/assets/60f62fcc-ece9-4c67-8b88-e226f8cad26a)
![{FF9D98EF-EF50-4994-BC3D-EC299875FDE6}](https://github.com/user-attachments/assets/db05a738-9030-429b-a913-33677dd57a98)

**Comment by fufesou** [comment]
@SerWax Thanks for your feedback.

> After about a day, LDAP logins fail with incorrect username/password, but no connections can be observer from the rustdesk pro server and the DCs.

I've read the code. The most possible reason is 

1. The LDAP configuration is Ok. Then you can connect with LDAP account.
2. The server is restarted, but it cannot connect LDAP server at the time.
3. Then you will face the error "Incorrect username or password".

Could you please confirm if the server was restarted?

**Comment by SerWax** [comment]
Hi @fufesou 

yes you are correct, the server was restarted.
Any chance the LDAP connection can be restored automatically at boot?

**Reply by fufesou** [reply]
> Any chance the LDAP connection can be restored automatically at boot?

No. It will only try to connect to the LDAP server once.

Sorry for the inconvenience.

This issue will be fixed in the next release.

**Reply by SerWax** [reply]
Understood, thank you
